### ISSUE TITLE
3.4 crashing El Capitan

### ISSUE THREAD

**Issue by connorjacobsen** [puma/puma#963]
Running Puma `3.4` with Rails `4.2.5.1` is causing my machine to crash when I start the server. Downgrading to Puma `2.14` fixes the issue. I'm running OS X `10.11.3`. Unfortunately, I don't know much else. We were able to reproduce this behavior multiple times on 4 different machines.

**Comment by evanphx** [comment]
It's causing the machine to crash? As in kernel panic?

**Comment by connorjacobsen** [comment]
Yes. I can dump the system log but there doesn't appear to anything useful there.

**Comment by connorjacobsen** [comment]
Just got it on `3.3` too.

**Comment by evanphx** [comment]
I really can not imagine what in Puma would be causing that.

**Comment by connorjacobsen** [comment]
`3.2` crashed it too. I have no idea what is going on.

**Comment by evanphx** [comment]
Sadly there is zero ability for me to reproduce this. If you have a way to reproduce it, I'll get right on it.

**Comment by fvosburgh** [comment]
I'm also getting this issue using ruby version `2.1.10`, rails version `4.1.0`, and puma version `3.4.0`. Downgrading to puma `2.14` fixed the issue as well.

**Comment by nolman** [comment]
Uninstalling all the previous versions of puma and reinstalling puma fixed this for me.

**Comment by evanphx** [comment]
@nolman I'm totally at a loss still. The kernel panic log doesn't indicate anything about why the ruby process crashed the kernel. Realistically, this has to be an OS X bug because any user process that can crash the kernel is a kernel bug. I know that's not useful to hear but I simply have no other ability to help you :(

**Comment by nolman** [comment]
Hey @evanphx so I have upgraded 2 boxes to the latest version of el capitan (`10.11.5`) and no longer get the kernel panics, I believe you are correct that it is a kernel bug. We just updated puma to `3.4.0` in our main codebase this morning, was going to wait a bit longer before updating the issue to ensure upgrading OSX really fixed the issue. Lets assume that the OSX update fixed it, if I see the issue again will comment again.
